tests/foreman/ui/test_computeresource_vmware.py

```diff
@@ -424,45 +360,55 @@ def test_positive_access_vmware_with_custom_profile(session, vmware):
+                    'provider_content.cluster': settings.vmware.cluster,
+                    'provider_content.annotation_notes': annotation_notes,
+                    'provider_content.virtual_hw_version': virtual_hw_version,
+                    'provider_content.firmware': firmware,
+                    'provider_content.resource_pool': resource_pool,
+                    'provider_content.folder': folder,
+                    'provider_content.memory_hot_add': memory_hot_add,
+                    'provider_content.cpu_hot_add': cpu_hot_add,
+                    'provider_content.cdrom_drive': cdrom_drive,
+                    'provider_content.storage': [value for value in storage_data.values()],
+                    'provider_content.network_interfaces': [
+                        value for value in network_data.values()
+                    ],
+                },
             )
-            for disk_index, expected_disk_value in enumerate(expected_controller_value['disks']):
-                provided_disk_value = {
-                    key: value
-                    for key, value in provided_controller_value['disks'][disk_index].items()
-                    if key in expected_disk_value
-                }
-                assert provided_disk_value == expected_disk_value
+            values = session.computeresource.read_computeprofile(cr_name, cprofile)
+            assert values['provider_content']['guest_os'] == guest_os_name
+            assert values['provider_content']['cpus'] == cpu
+            assert values['provider_content']['memory'] == memory
+            assert values['provider_content']['cluster'] == settings.vmware.cluster
+            assert values['provider_content']['annotation_notes'] == annotation_notes
+            assert values['provider_content']['virtual_hw_version'] == virtual_hw_version
+            if not is_open('BZ:2266672'):
+                assert values['provider_content']['firmware'] == firmware
+            assert values['provider_content']['resource_pool'] == resource_pool
+            assert values['provider_content']['folder'] == folder
+            assert values['provider_content']['memory_hot_add'] == memory_hot_add
+            assert values['provider_content']['cpu_hot_add'] == cpu_hot_add
+            assert values['provider_content']['cdrom_drive'] == cdrom_drive
+            assert values['provider_content']['storage'][0]['controller'] == VMWARE_CONSTANTS.get(
+                'scsicontroller'
+            )
+            assert values['provider_content']['storage'][0]['disks'][0]['size'] == disk_size
+            assert values['provider_content']['network_interfaces'][0][
+                'nic_type'
+            ] == VMWARE_CONSTANTS.get('network_interface_name')
+            assert values['provider_content']['network_interfaces'][0]['network'] == network
+        session.computeresource.delete(cr_name)
+        assert not session.computeresource.search(cr_name)
```

Comment:
Can we add a step to delete computeprofile here as well? and do you think it'd make sense to add request.addfinalizer to remove this CR in case test fails earlier?

Reply:
Added the request.addfinalizer, we cannot delete computeprofiles as there are three by default.

Reply:
@shubhamsg199 Not like this, we can delete via UI at end as it was earlier, but if in case it fails before reaching to it, we could have finalizer method which could check if it exists then only delete it in teardown

Reply:
Updated

Reply:
- We're still not covering UI method to delete CR in end after update CP
- finalizer method shouldn't be at last, as our main motive is if update CP or any step fails before then it should run, and keeping it at last wouldn't run this teardown part